test/fuzz/parser_fuzzers.go

```diff
@@ -0,0 +1,248 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fuzz
+
+import (
+	"io"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/client-go/util/keyutil"
+	"k8s.io/code-generator/cmd/client-gen/generators/util"
+	envutil "k8s.io/kubectl/pkg/cmd/set/env"
+	"k8s.io/kubectl/pkg/util/certificate"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	"k8s.io/kubernetes/pkg/credentialprovider"
+	"k8s.io/kubernetes/pkg/kubelet/cm"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+	"k8s.io/kubernetes/pkg/kubelet/eviction"
+	"k8s.io/kubernetes/pkg/registry/core/service/portallocator"
+	"k8s.io/kubernetes/pkg/util/parsers"
+
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+)
+
+func FuzzParseSelector(data []byte) int {
+	_, _ = fields.ParseSelector(string(data))
+	return 1
+}
+
+func FuzzParseQuantity(data []byte) int {
+	_, _ = resource.ParseQuantity(string(data))
+	return 1
+}
+
+func FuzzParseGroupVersion(data []byte) int {
+	_, _ = schema.ParseGroupVersion(string(data))
+	return 1
+}
+
+func FuzzParseResourceArg(data []byte) int {
+	_, _ = schema.ParseResourceArg(string(data))
+	return 1
+}
+
+func FuzzLabelsParse(data []byte) int {
+	_, _ = labels.Parse(string(data))
+	return 1
+}
+
+func FuzzParseImageName(data []byte) int {
+	_, _, _, _ = parsers.ParseImageName(string(data))
+	return 1
+}
+
+func FuzzParsePrivateKeyPEM(data []byte) int {
+	_, _ = keyutil.ParsePrivateKeyPEM(data)
+	return 1
+}
+
+func FuzzParsePublicKeysPEM(data []byte) int {
+	_, _ = keyutil.ParsePublicKeysPEM(data)
+	return 1
+}
+
+func FuzzParseCSR(data []byte) int {
+	_, _ = certificate.ParseCSR(data)
+	return 1
+}
+
+func FuzzMeta1ParseToLabelSelector(data []byte) int {
+	_, _ = metav1.ParseToLabelSelector(string(data))
+	return 1
+}
+
+func FuzzParseHostPort(data []byte) int {
+	_, _, _ = kubeadmutil.ParseHostPort(string(data))
+	return 1
+}
+
+func FuzzParseVersion(data []byte) int {
+	f := fuzz.NewConsumer(data)
+	vString1, err := f.GetString()
+	if err != nil {
+		return 0
+	}
+	v1, err := version.ParseSemantic(vString1)
+	if err != nil {
+		return 0
+	}
+	_ = v1.String()
+	vString2, err := f.GetString()
+	if err != nil {
+		return 0
+	}
+	v2, err := version.ParseGeneric(vString2)
+	if err != nil {
+		return 0
+	}
+	_ = v1.AtLeast(v2)
+	return 1
+}
+
+func FuzzCreatePortAllocator(data []byte) int {
+	pr, err := net.ParsePortRange(string(data))
+	if err != nil {
+		return 0
+	}
+	r, err := portallocator.NewInMemory(*pr)
+	if err != nil {
+		return 0
+	}
+	for i := 0; i < 20; i++ {
+		_, err = r.AllocateNext()
```

Comment:
This constant depend on the range, i.e something with a range with less than 20 will return an error

Reply:
This is merely an issue if the fuzzer would crash with false positives which it doesn't.

Reply:
TIL, thanks